Framework/Geometry/src/Instrument/ParameterMap.cpp

```diff
@@ -1152,14 +1152,20 @@ ParameterMap::create(const std::string &className,
   return ParameterFactory::create(className, name);
 }
 
-/// Only for use by ExperimentInfo. Returns returns true if this instrument
-/// contains a DetectorInfo.
-bool ParameterMap::hasDetectorInfo() const {
+/** Only for use by ExperimentInfo. Returns returns true if this instrument
+ contains a DetectorInfo.
+
+ The `instrument` argument is needed for the special case of having a neutronic
+ *and* a physical instrument. `Instrument` uses the same parameter map for both,
+ but the DetectorInfo is only for the neutronic instrument. */
+bool ParameterMap::hasDetectorInfo(const Instrument *instrument) const {
+  if (instrument != m_instrument)
```

Comment:
Would be good to add a comment around `m_instrument` in the header regarding the fact that `m_instrument` is taken to be the neutronic one.